src/libpmempool/replica.c

```diff
@@ -561,31 +756,103 @@ check_badblocks(struct pool_set *set, struct poolset_health_status *set_hs)
 
 		for (unsigned p = 0; p < rep->nparts; ++p) {
 			const char *path = PART(rep, p)->path;
+			struct part_health_status *part_hs = &rep_hs->part[p];
 
 			int exists = os_access(path, F_OK) == 0;
 			if (!exists)
 				continue;
 
-			int ret = os_badblocks_check_file(path);
+			int ret = redo_log_exist(path, part_hs);
 			if (ret < 0) {
-				ERR(
-					"checking replica for bad blocks failed -- '%s'",
+				ERR("checking redo log failed -- '%s'",
 					path);
 				return -1;
 			}
 
-			if (ret > 0) {
-				/* bad blocks were found */
-				rep_hs->part[p] |= IS_BROKEN;
-				rep_hs->flags |= IS_BROKEN;
+			if (read_redo_logs) {
```

Comment:
move this to a separate function. I see it this way:
VEC(struct badblock) replica_part_badblocks_get(); - which would return an array of badblocks (might be empty), taking the redo log stuff into account
replica_part_badblocks_clear(VEC(struct badblock) blocks); - which would take the array and clear it.